.github/workflows/on-change.yaml

```diff
@@ -0,0 +1,27 @@
+---
+name: "Syncron"
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Clone repository"
+        uses: actions/checkout@v3
+      - name: "Setup Go"
+        uses: actions/setup-go@v3
+        with:
+          go-version: ">=1.17.0"
+      - name: "Run linter - StaticCheck"
+        uses: dominikh/staticcheck-action@v1.2.0
+        with:
+          version: "2022.1"
+          install-go: false
+      - name: "Build binary - Server"
+        run: go build -o .go/builds/server cmd/server.go
+      - name: "Run tests - All"
+        run: go test ./...
```

Comment:
'go test .' won't work?

Reply:
That would just look for tests on the root of the repository. I need to add the three dots to make it recursive.

Reply:
https://blog.riff.org/2014_09_27_go_tip_of_the_day_running_tests_for_all_subpackages_recursively